libs/aws/langchain_aws/logger_util.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import os
+
+# Environment variable to set the application logger(s) in debug mode during runtime
+LANGCHAIN_AWS_DEBUG: str = os.environ.get("LANGCHAIN_AWS_DEBUG", "false")
+__DEBUG: bool = True if LANGCHAIN_AWS_DEBUG.lower() in ["true", "1"] else False
+
+# Flag for root debug logger to set the root debug logger in debug mode during runtime
+# Root debug logger will print boto3 as well as application debug logs if set to true
+# This flag will be set to true if LANGCHAIN_AWS_DEBUG = LANGCHAIN_AWS_DEBUG_ROOT = true
+__LANGCHAIN_AWS_DEBUG_ROOT: str = os.environ.get("LANGCHAIN_AWS_DEBUG_ROOT", "false")
+__ROOT_DEBUG: bool = __DEBUG if __LANGCHAIN_AWS_DEBUG_ROOT.lower() \
+    in ["true", "1"] else False
+
+# If application level debug flag is set, set the default logging level to DEBUG 
+# else ERROR
+if __DEBUG:
+    DEFAULT_LOG_LEVEL: int = logging.DEBUG
+else:
+    DEFAULT_LOG_LEVEL: int = logging.ERROR
+
+#  Checking if we have a log file handler available, if not, setting default handler 
+#  to None
+DEFAULT_LOG_FILE: str = os.environ.get("LANGCHAIN_AWS_LOG_OUTPUT")
+if DEFAULT_LOG_FILE:
+    DEFAULT_LOG_HANDLER: logging.Handler = logging.FileHandler(DEFAULT_LOG_FILE, "a")
+else:
+    DEFAULT_LOG_HANDLER: logging.Handler = None
+
+# If application debug mode set then include filename and line numbers in the log format
+if __DEBUG:
+    DEFAULT_LOG_FORMAT: str = (
+        "%(asctime)s %(levelname)s | [%(filename)s:%(lineno)s] | %(name)s - %(message)s"
+    )
+else:
+    DEFAULT_LOG_FORMAT: str = "%(asctime)s %(levelname)s | %(name)s - %(message)s"
+
+
+# This block is to set a log formatter. colorama is used to print colored logs
+try:
+    import colorama
+    import coloredlogs
+
+    colorama.init()
+
+    # If a DEFAULT_LOG_HANDLER is set in the LANGCHAIN_AWS_LOG_OUTPUT env var, skip
+    # colored logs format
+    if DEFAULT_LOG_HANDLER:
+        DEFAULT_LOG_FORMATTER: logging.Formatter = logging.Formatter(DEFAULT_LOG_FORMAT)
+    else:
+        DEFAULT_LOG_FORMATTER: logging.Formatter = coloredlogs.ColoredFormatter(
+            DEFAULT_LOG_FORMAT
+        )
+# If colorama is not present, use the logging lib formatter without colored logs
+except ImportError:
+    colorama = None
+    coloredlogs = None
+
+    DEFAULT_LOG_FORMATTER: logging.Formatter = logging.Formatter(DEFAULT_LOG_FORMAT)
```

Comment:
IMO we shouldn't fail this silently - add an logger statement in case users are interested in enabling coloring